electricitymap/contrib/capacity_parsers/IRENA.py

```diff
@@ -0,0 +1,146 @@
+    )
+    url_year = datetime.now().year
+    filename_combinations = [
+        f"ELECCAP_{url_year}_cycle2.px",
+        f"ELECCAP_{url_year}_cycle1.px",
+        f"ELECCAP_{url_year}.px",
+        f"ELECCAP_{url_year-1}_cycle2.px",
+        f"ELECCAP_{url_year-1}_cycle1.px",
+        f"ELECCAP_{url_year-1}.px",
+    ]
+    json_query = {
+        "query": [
+            {
+                "code": "Year",
+                "selection": {
+                    "filter": "item",
+                    "values": [target_datetime.strftime("%y")],
+                },
+            }
+        ],
+        "response": {"format": "json"},
+    }
+    data = None
+    for filename in filename_combinations:
+        url = base_url + filename
+
+        json_data = json.dumps(json_query)
+        r: Response = Session().post(url, data=json_data)
+        if r.status_code == 200:
+            data = r.json()
+        else:
+            continue
+    if not data:
+        raise ValueError(f"Could not fetch data for {target_datetime.year}")
+    return data["data"]
+
+
+def reallocate_capacity_mode(zone_key: ZoneKey, mode: int) -> dict:
+    if zone_key in SPECIFIC_MODE_MAPPING:
+        if mode in SPECIFIC_MODE_MAPPING[zone_key]:
+            return SPECIFIC_MODE_MAPPING[zone_key][mode]
+    else:
+        return IRENA_JSON_TO_MODE_MAPPING[mode]
+
+
+def get_capacity_data_for_all_zones(target_datetime: datetime):
+    data = get_data_from_url(target_datetime)
+    capacity_dict = {}
+    for item in data:
+        if pycountry.countries.get(alpha_3=item["key"][0]) is not None:
+            zone = pycountry.countries.get(alpha_3=item["key"][0]).alpha_2
+        else:
+            pass
+        mode: str = IRENA_JSON_TO_MODE_MAPPING[int(item["key"][1])]
+        value: float = round(float(item["values"][0]), 0)
+        datetime_value: datetime = datetime.strptime(item["key"][-1], "%y")
+
+        if zone not in capacity_dict:
+            zone_dict = {
+                mode: {
+                    "datetime": datetime_value,
+                    "value": value,
+                }
+            }
+            capacity_dict[zone] = zone_dict
+        else:
+            mode = reallocate_capacity_mode(zone, int(item["key"][1]))
+            if mode in capacity_dict[zone]:
+                zone_dict = capacity_dict[zone][mode]
+                capacity_dict[zone][mode]["value"] += value
+            else:
+                capacity_dict[zone] = {
+                    **capacity_dict[zone],
+                    **{
+                        mode: {
+                            "datetime": datetime_value,
+                            "value": value,
+                        }
+                    },
+                }
+    return capacity_dict
+
+
+def fetch_production_capacity(target_datetime: datetime, zone_key: ZoneKey) -> dict:
+    all_capacity = get_capacity_data_for_all_zones(target_datetime)
```

Comment:
Minor suggestion: In the beginning I didn't understand why we were getting capacity data for all zones instead of just querying the zone we wanted. Now I get it. Would it make sense to include a comment that explains why we get data for all zones (the .px data has data for all zones in one file) and then we select what we want?

Reply:
there are two ways of doing it actually, we could pass the zone param in the web query and fetch zone by zone but i think this is slightly more efficitent as we only do one query and then filter it

Reply:
makes sense